app/models/tag.rb

```diff
@@ -200,7 +200,7 @@ def find_or_create_by_name(name, creator: CurrentUser.user)
             # next few lines if the category is changed.
             tag.update_category_cache
 
-            if Pundit.policy!(creator, tag).can_change_category?
+            if Pundit.policy!(creator, tag).can_change_category? && !(category_id == 6 && tag.post_count > 0 && !creator.is_admin?)
```

Comment:
This usage of the number 6 should be some kind of self-commenting variable instead of what is otherwise just a random number.

`TagCategory.mapping["deprecated"]` should work, but perhaps the `TagCategory` module could use a new enum so that hash representation doesn't have to be used. At one point in time the TagCategory module had such a thing, but it appears to have been removed at some point in the past.